pssh/clients/base/single.py

```diff
@@ -302,8 +302,12 @@ def _connect_socket(self, family, _type, proto, sock_addr, host, port, retries):
         try:
             self.sock.connect(sock_addr)
         except ConnectionRefusedError:
+            self.sock.shutdown(socket.SHUT_RDWR)
```

Comment:
```suggestion
            if not self.sock.closed:
                self.sock.shutdown(socket.SHUT_RDWR)
                self.sock.close()
```

Otherwise will get test failures for already closed sockets - see test failures.